### PR TITLE
Task 7

### PR DESCRIPTION
**Pull request by SamoylenkVi**

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
-import SiteMenuView from './view/menu';
 import PagePresenter from './presenter/page';
 import { renderElement } from './utils/render';
 import { generateCardFilm } from './mock/card-film';
+import FilmsModel from './model.js/films';
+import CommentsModel from './model.js/comments';
 import { FILM_COUNT, RenderPosition } from './constants';
+import allComments from './mock/film-comments';
 
 const filmCards = new Array(FILM_COUNT)
```
Comment (Pavel-Liteiniy) [inline src/main.js]:
```suggestion
const filmCards = Array.from({ length: FILM_COUNT }, (_, index) => generateCardFilm(idElement))
```

File: src/model.js/comments.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import Observer from '../utils/observer';
```
Comment (Pavel-Liteiniy) [inline src/model.js/comments.js]: Incorrect folder name `model.js`

File: src/model.js/comments.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+export default class Comments extends Observer {
+  constructor() {
+    super();
+
+    this._comments = [];
```
Comment (Pavel-Liteiniy) [inline src/model.js/comments.js]:
```suggestion
    this._comments = {};
```

Array or Object?

File: src/model.js/films.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    } else {
+      this._filteredAndSortedFilms = this._filteredFilms.slice();
+    }
+  }
+
+  get filteredAndSortedFilms() {
+    return this._filteredAndSortedFilms;
+  }
+
+  set currentFilterType(filterType) {
```
Comment (Pavel-Liteiniy) [inline src/model.js/films.js]:
```suggestion
  set filterType(filterType) {
```

Put getter and setter together with the same name

File: src/presenter/movie-list.js [hunk 5]
```diff
@@ -2,100 +2,183 @@ import EmptyFilmMessageView from '../view/message-empty-film-list';
...
+
+    this._renderFilms(films);
+
+    if (filmsCount > this._renderedTaskCount) {
       this._renderLoadMoreButton();
     }
   }
 
+  _renderFilms(films) {
+    films.forEach((film) => this._renderFilmCard(film));
```
Comment (Pavel-Liteiniy) [inline src/presenter/movie-list.js:147]:
```suggestion
    films.forEach(this._renderFilmCard);
```

File: src/presenter/movie-list.js [hunk 6]
```diff
@@ -2,100 +2,183 @@ import EmptyFilmMessageView from '../view/message-empty-film-list';
...
+  }
+
   _renderFilmCard(film) {
     const filmPresenter = new MovieCardPresenter(
```
Comment (Pavel-Liteiniy) [inline src/presenter/movie-list.js]:
```suggestion
    const movieCardPresenter = new MovieCardPresenter(
```

**Review by Pavel-Liteiniy — CHANGES_REQUESTED** [review]

File: src/main.js (hunk 1 above)
Comment (SamoylenkVi) [reply src/main.js]: fixed

File: src/model.js/comments.js (hunk 2 above)
Comment (SamoylenkVi) [reply src/model.js/comments.js]: fixed

File: src/model.js/comments.js (hunk 3 above)
Comment (SamoylenkVi) [reply src/model.js/comments.js]: fixed

File: src/model.js/films.js (hunk 4 above)
Comment (SamoylenkVi) [reply src/model.js/films.js]: fixed

File: src/presenter/movie-list.js (hunk 5 above)
Comment (SamoylenkVi) [reply src/presenter/movie-list.js:147]: fixed

File: src/presenter/movie-list.js (hunk 6 above)
Comment (SamoylenkVi) [reply src/presenter/movie-list.js]: fixed

**Comment by Pavel-Liteiniy** [comment]
approved

**Comment by Pavel-Liteiniy** [comment]
image

**Pull request merged**